freelist.go

```diff
@@ -24,6 +24,7 @@ type pidSet map[common.Pgid]struct{}
 type freelist struct {
 	freelistType   FreelistType                              // freelist type
 	ids            []common.Pgid                             // all free and available free page ids.
+	readonlyTXIDs  []common.Txid                             // all readonly transaction IDs.
```

Comment:
do we really need this? this seems tracked in the `pending` map already

Reply:
The TXIDs included in pending are actually writing TXIDs.

freelist is one of the most sensitive & important parts of bbolt, let's do it step by step. We need to ensure each step is well understood.